apiserver/admin.go

```diff
@@ -101,7 +101,7 @@ func (a *admin) login(req params.LoginRequest, loginVersion int) (params.LoginRe
 	}
 
 	// Fetch the API server addresses from state.
-	hostPorts, err := a.root.state.APIHostPorts()
```

Comment:
given that this one is being returned in the params.LoginResult(), don't we need to have a check somewhere that sees that this is a Client login vs an Agent login and replaces the contents with APIHostPortsForClients ?

Have you checked that "juju bootstrap" to a machine with multiple IP addresses leaves ~/.local/share/juju/controllers.yaml with all addresses specified. And that doing "juju status" on that same controller continues to list all addresses?

That also indicates we need a Unit test around the right thing happening here.

Reply:
Yes. I checked the API addresses with "juju status", which lists all the IPs. They are present in _controllers.yaml_.

Looking at _params.LoginRequest_, it looks like the tag kind might be used to determine the caller. What kinds denote a client login?

Reply:
Client login should only be User, AIUI